design/ip-filtering-design.md

```diff
@@ -0,0 +1,140 @@
+# IP Filtering 
+
+Status: Draft
+
+## Abstract
+
+Allow or deny traffic to services behind contour with an IP allowlist or an IP denylist.
+
+## Background
+
+There have been [long standing requests](https://github.com/projectcontour/contour/issues/62) to support some form of IP filtering.
+IP filtering is a useful tool for network segregation that is supported by other ingress controllers like [ingress-nginx](https://github.com/kubernetes/ingress-nginx/blob/87d1b8bbf28265386b339e65d8943d8c3f8582ed/docs/user-guide/nginx-configuration/annotations.md#whitelist-source-range) and [ambassador](https://www.getambassador.io/docs/edge-stack/latest/topics/running/ambassador#ip-allow-and-deny), and is a [common](https://github.com/projectcontour/contour/issues/62#issuecomment-698141472) [request](https://github.com/projectcontour/contour/issues/62#issuecomment-843193956) among the community, often listed as a [blocker](https://github.com/projectcontour/contour/issues/3693#issuecomment-1317049708) for adopting contour.
+
+## Goals
+
+- Support ip allowlists and denylists via Contour APIs.
+- Support filtering on request IP or on forwarded IP (i.e. via `X-Forwarded-For`)
+
+## Non Goals
+- Define Gateway API support for ip filtering. This is an obvious next step, but requires broader agreement.
```

Comment:
we'll likely need to write up a design for a CR to be used as an extension Filter or Policy for Gateway API

Reply:
Are you saying that you think we should bring the Gateway API into scope? Or just that that's the likely direction if/when this is approached in for Gateway API?

Reply:
a bit of both actually, probably good to get a baseline of what we want to support done in HTTPProxy as well as acknowledging that unless the upstream Gateway API project supports it natively we will have to write our own CR to handle that feature